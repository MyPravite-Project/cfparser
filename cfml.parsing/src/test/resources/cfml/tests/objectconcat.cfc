x = {
    name: rc.ticket.getShortId()
    ,action: 'support:tickets/view/id/' & rc.ticket.getShortId()
};