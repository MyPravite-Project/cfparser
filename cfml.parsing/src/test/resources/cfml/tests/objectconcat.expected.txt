/*===TOKENS===*/
IDENTIFIER               <x>
'='                      <=>
'{'                      <{>
Hidden:NEWLINE                  <>
IDENTIFIER               <name>
':'                      <:>
IDENTIFIER               <rc>
'.'                      <.>
IDENTIFIER               <ticket>
'.'                      <.>
IDENTIFIER               <getShortId>
'('                      <(>
')'                      <)>
Hidden:NEWLINE                  <>
','                      <,>
IDENTIFIER               <action>
':'                      <:>
OPEN_STRING              <'>
STRING_LITERAL           <support:tickets/view/id/>
CLOSE_STRING             <'>
'&'                      <&>
IDENTIFIER               <rc>
'.'                      <.>
IDENTIFIER               <ticket>
'.'                      <.>
IDENTIFIER               <getShortId>
'('                      <(>
')'                      <)>
Hidden:NEWLINE                  <>
'}'                      <}>
';'                      <;>
/*===TREE===*/
(scriptBlock
  (element
    (statement
      (assignmentExpression
        (startExpression
          (baseExpression (unaryExpression (memberExpression (identifier x))))
        )
        =
        (startExpression
          (baseExpression
            (unaryExpression
              (primaryExpression
                (implicitStruct
                  {
                  (implicitStructElements
                    (implicitStructExpression
                      (implicitStructKeyExpression (multipartIdentifier (identifier name)))
                      :
                      (baseExpression
                        (unaryExpression
                          (memberExpression
                            (identifier rc)
                            .
                            (identifier ticket)
                            .
                            (qualifiedFunctionCall (identifier getShortId) ( argumentList ))
                          )
                        )
                      )
                    )
                    ,
                    (implicitStructExpression
                      (implicitStructKeyExpression (multipartIdentifier (identifier action)))
                      :
                      (baseExpression
                        (baseExpression
                          (unaryExpression
                            (primaryExpression
                              (literalExpression
                                (stringLiteral ' (stringLiteralPart support:tickets/view/id/) ')
                              )
                            )
                          )
                        )
                        &
                        (baseExpression
                          (unaryExpression
                            (memberExpression
                              (identifier rc)
                              .
                              (identifier ticket)
                              .
                              (qualifiedFunctionCall (identifier getShortId) ( argumentList ))
                            )
                          )
                        )
                      )
                    )
                  )
                  }
                )
              )
            )
          )
        )
      )
      (endOfStatement ;)
    )
  )
)
/*======*/
/*===DECOMPILE===*/
x = {name:rc.ticket.getShortId(),action:'support:tickets/view/id/' & rc.ticket.getShortId()}
/*======*/